### PR TITLE
Workflow modifications for official pipeline release

### PR DESCRIPTION
**Pull request by ReinierCC**
Added:

- Hash Version for azure/login action.
- Changelog step back
- New step to find the vsix file
- Step to add vsix file to the release back.

Tested the operation of finding the vsix on local fork:
Screenshot 2024-10-25 at 12 02 19 PM

Using npx for vsce command for one off package and publish. Reference: https://docs.npmjs.com/cli/v9/commands/npx?v=true

### CODE REVIEW

**Review by hsubramanianaks — DISMISSED** [review]

File: .github/workflows/publish-v2.yml [hunk 1]
```diff
@@ -86,7 +86,7 @@ jobs:
         run: npx vsce package
 
       - name: federated login
-        uses: azure/login@v2
+        uses: azure/login@a65d910e8af852a8061c627c456678983e180302 # v2.2.0
```
Comment (Tatsinnit) [inline .github/workflows/publish-v2.yml:89]: 💡 Can we please add the cheangelog and the release part back into this :) from the current `publics.yaml` we can rename this file latter once we retire current release process.

thank you

**Review by Tatsinnit — COMMENTED** [review]
❤️ Just a note please: Sorry, I forgot to add samples here in my other comment. post line 49 of this file I think we will add in:

I think here are the 2 missing pieces, I am sure you will take one fine look in case I missed anything please.

```*.yml
- name: Check version is mentioned in Changelog
        id: changelog_reader
        uses: mindsers/changelog-reader-action@32aa5b4c155d76c94e4ec883a223c947b2f02656 # v2.2.3
        with:
            validation_depth: 10
            version: ${{ steps.package_version.outputs.current-version }}
            path: 'CHANGELOG.md'
```

and then this: (Just near second job where we `npm install` 


```*.yml
- name: Create a Release
        id: create_release
        uses: actions/create-release@0cb9c9b65d5d1901c1f53e5e66eaf4afd303e70e # v1.1.4
        env:
            GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} 
        with:
            tag_name : ${{ needs.JobBuild.outputs.currentversion }}
            release_name: ${{ needs.JobBuild.outputs.currentversion}}
            body: Publish ${{ needs.JobBuild.outputs.changelog_reader_changes }}
```

**Comment by ReinierCC** [comment]
Updating based on discussion with @Tatsinnit

**Review by Tatsinnit — APPROVED** [review]
❤️ Thank you so much for this. Looks good.

**Review by hsubramanianaks — APPROVED** [review]
looks good to me, I hope this was tested in forked branch. Thank you @ReinierCC and @Tatsinnit

**Pull request merged**